config/config-istio.yaml

```diff
@@ -44,6 +44,11 @@ data:
     # used pre-0.3.
     gateway.knative-ingress-gateway: "istio-ingressgateway.istio-system.svc.cluster.local"
 
+    # The default namespace of the gateway is the same with the serving
+    # system. The users can also use gateway in a different namespace
+    # by explicitly specifying the custom namespace.
+    gateway.custom-namespace.custom-ingress-gateway: "my-ingress.my-namespace.svc.cluster.local"
```

Comment:
Can we use the examples as defaults? Although it is not fully possible, https://github.com/knative/serving/issues/3492 tries to do that.
Then, we can add general syntax example like `gateway.{{gateway_namespace}}.{{gateway_name}}: "istio-ingressgateway.istio-system.svc.cluster.local"` to the explanation.

Reply:
In this case, all gateway settings should explicitly specify the namespace including those in the serving namespace. This sounds good to me. @trshafer WDYT?

Reply:
@tcnghia Hi, what do you think about this? Thank you.

Reply:
Just in case, I meant L#50 could be `gateway.knative-serving.knative-ingress-gateway: "istio-ingressgateway.istio-system.svc.cluster.local"`, which is same value with default [here](https://github.com/knative/serving/blob/a270a7bd0008f0c853aefcd9a0b8a9666557e862/pkg/reconciler/ingress/config/istio.go#L50).

Then, we can add the explanation like `The format must be gateway.{{gateway_namespace}}.{{gateway_name}}: "istio-ingressgateway.istio-system.svc.cluster.local". The {{gateway_namespace}} is an optional, when you omit it blah-blah-blah`.

Reply:
I'm not sure which case is more common among users: using the default `knative-ingress-gateway` or custom gateway. If the default gateway is more often used, maybe we should first explain the default setting, and then describe the option to customize. Otherwise I think your idea makes more sense.

Reply:
I'm sure that the default `knative-ingress-gateway` is more common. What I wanted to say is that L#50 should be default as suggested by https://github.com/knative/serving/issues/3492.
Let's say _if_ we add the unit test proposed in https://github.com/knative/serving/issues/3492:

> Add unit tests that ensure that the default values in the code match the values in the _example section.

The unit test fails with current `custom-namespace`, doesn't it?

Reply:
Sorry I misunderstood your suggestions before. Updated the config examples. Thanks.